### PR TITLE
[5.5] Collection as parameter on Collection::except 

### PR DESCRIPTION
**Pull request by glorand**
```php
$collectionA = collect(['first' => 'First', 'last' => 'Last']);
$collectionB = collect(['first' => 'First']);
#Instead of 
$collectionC = $collectionA->except($collectionB->keys()->all());
#you can use in this way
$collectionC = $collectionA->except($collectionB);
```

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -385,12 +385,16 @@ public function every($key, $operator = null, $value = null)
...
      *
-     * @param  mixed  $keys
+     * @param  mixed|Collection  $keys
      * @return static
      */
     public function except($keys)
     {
-        $keys = is_array($keys) ? $keys : func_get_args();
+        if ($keys instanceof Collection) {
+            $keys = $keys->keys()->items;
```
Comment (taylorotwell) [inline src/Illuminate/Support/Collection.php]: Isn't items protected?

Comment (lagbox) [reply src/Illuminate/Support/Collection.php]: yes, it has direct access to it, but i would suppose the method call would be better?

Comment (glorand) [reply src/Illuminate/Support/Collection.php]: @lagbox *all()* method? probably yes

**Comment by taylorotwell** [comment]
Reverting this sense it doesn't make any sense.

**Pull request merged**